MapleServer2/Commands/Game/GMCommands.cs

```diff
@@ -294,3 +297,223 @@ public override void Execute(GameCommandTrigger trigger)
         ShopHelper.OpenSystemShop(trigger.Session, 999999, 29000307);
     }
 }
+
+public class TimeScaleCommand : InGameCommand
+{
+    public TimeScaleCommand()
+    {
+        Aliases = new()
+        {
+            "timescale"
+        };
+        Description = "Sets the time scale.";
+        Parameters = new()
+        {
+            new Parameter<byte>("enable", "Enabled, e.g.: 1 (true) / 0 (false)"),
+            new Parameter<float>("startScale", "Start scale, e.g.: 0.0 ~ 1.0"),
+            new Parameter<float>("endScale", "End scale, e.g.: 0.0 ~ 1.0"),
+            new Parameter<float>("duration", "Duration, e.g.: 10.0 (seconds)"),
+            new Parameter<byte>("interpolator", "Enable fade between start and end scale, e.g.: 1 (true) / 0 (false)"),
+            new Parameter<byte>("broadcast", "Broadcast to all players in the field, e.g.: 1 (true) / 0 (false)")
+        };
+        Usage = "/timescale [enable] [startScale] [endScale] [duration] [interpolator] [broadcast]";
+    }
+
+    public override void Execute(GameCommandTrigger trigger)
+    {
+        bool enable = trigger.Get<byte>("enable") == 1;
+        if (!enable)
+        {
+            trigger.Session.Send(TimeScalePacket.SetTimeScale(false, 0, 0, 0, 0));
+            return;
+        }
+
+        float startScale = trigger.Get<float>("startScale");
+        float endScale = trigger.Get<float>("endScale");
+        float duration = trigger.Get<float>("duration");
+        byte interpolator = trigger.Get<byte>("interpolator");
+        bool broadcast = trigger.Get<byte>("broadcast") == 1;
+        PacketWriter packet = TimeScalePacket.SetTimeScale(true, startScale, endScale, duration, interpolator);
+        if (broadcast)
+        {
+            trigger.Session.FieldManager.BroadcastPacket(packet);
+            return;
+        }
+
+        trigger.Session.Send(packet);
+    }
+}
+
+public class WeatherCommand : InGameCommand
+{
+    private readonly string WeatherTypes = "";
+
+    public WeatherCommand()
+    {
+        foreach (object slot in Enum.GetValues(typeof(WeatherType)))
+        {
+            WeatherTypes += $"{slot}, ";
+        }
+
+        WeatherTypes = WeatherTypes.TrimEnd(',', ' ');
+
+        Aliases = new()
+        {
+            "weather"
+        };
+        Description = "Sets the weather.";
+        Parameters = new()
+        {
+            new Parameter<string>("weatherType", $"Weather type, e.g.: Rain ({WeatherTypes})"),
+        };
+        Usage = "/weather [weatherType]";
+    }
+
+    public override void Execute(GameCommandTrigger trigger)
+    {
+        string weather = trigger.Get<string>("weatherType");
+
+        if (!Enum.TryParse(weather, out WeatherType weatherType))
+        {
+            trigger.Session.SendNotice($"Available weathers: {WeatherTypes}");
+            return;
+        }
+
+        trigger.Session.Send(FieldPropertyPacket.ChangeWeather(weatherType));
```

Comment:
Should these weather/color/etc packets be sent to all players in the field as well?

Reply:
Updated